tools/tungsten/core/src/com/google/android/filament/tungsten/texture/TextureUtils.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.filament.tungsten.texture
+
+import com.google.android.filament.Engine
+import com.google.android.filament.Texture
+
+import javax.imageio.ImageIO
+import java.awt.image.BufferedImage
+import java.awt.image.DataBufferByte
+import java.io.File
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+
+private fun Texture.InternalFormat.toSrgb(): Texture.InternalFormat {
+    return when (this) {
+        Texture.InternalFormat.RGB8 -> Texture.InternalFormat.SRGB8
+        Texture.InternalFormat.RGBA8 -> Texture.InternalFormat.SRGB8_A8
+        else -> this
+    }
+}
+
+object TextureUtils {
+
+    fun createDefaultTexture(engine: Engine): Texture {
+        val stream = javaClass.classLoader.getResourceAsStream("checkerboard.png")
+        val texture = stream?.let { s ->
+            ImageIO.read(s)?.let {
+                img -> loadTextureFromImage(engine, img)
+            }
+        }
+        return texture ?: throw RuntimeException("Could not load default texture")
+    }
+
+    fun loadTexture(engine: Engine, file: File): Texture? {
+        val img = try {
+            ImageIO.read(file) ?: return null
+        } catch (e: IOException) {
+            System.err.println("Could not read image ${file.canonicalPath}.")
+            e.printStackTrace()
+            return null
+        }
+
+        return loadTextureFromImage(engine, img)
+    }
+
+    /**
+     * Based on the number of components the image has and its color space, decide which
+     * texture formats to use.
+     */
+    private fun decideTextureFormat(
+        img: BufferedImage
+    ): Pair<Texture.InternalFormat, Texture.Format> {
+        val isSrgb = img.colorModel.colorSpace.isCS_sRGB
```

Comment:
That's a good first step but we need another way to sample an sRGB image as non sRGB. sRGB is assumed to be the default profile when missing, so this check might not always be correct.